src/__tests__/extractMeta.js

```diff
@@ -667,18 +667,46 @@ describe('Metadata extractor', () => {
         const extracted = extractMetadata(props);
 
         expect(extracted).toEqual({
-            'Ui label': {
+            devName: {
                 label: 'Ui label',
                 propName: 'devName',
                 type: 'string',
                 isPrivate: true,
                 tooltip: 'A tooltip'
             },
-            'Ui label two': {
+            devNameTwo: {
                 label: 'Ui label two',
                 propName: 'devNameTwo',
                 type: 'string'
             }
         });
     });
+
+    it('Extracts metadata safely if labels are not unique', () => {
+        const props = {
+            devName: {
+                type: ElementPropTypes.string,
+                label: 'Ui label'
+            },
+            devNameTwo: {
+                type: ElementPropTypes.string,
+                label: 'Ui label'
+            }
+        };
+
+        const extracted = extractMetadata(props);
+
+        expect(extracted).toEqual({
+            devName: {
+                label: 'Ui label',
+                propName: 'devName',
+                type: 'string'
+            },
+            devNameTwo: {
+                label: 'Ui label',
+                propName: 'devNameTwo',
+                type: 'string'
+            }
+        });
+    });
```

Comment:
@divinentd I wrote up an example of the shape props for testing, but I'm not sure how it's supposed to look extracted:

```js
    const props = {
        arrayOne: {
            label: 'Array label',
            type: ElementPropTypes.arrayOf(
                ElementPropTypes.shape({
                    keyOne: {
                        label: 'Ui label one',
                        type: ElementPropTypes.string
                    },
                    keyTwo: {
                        label: 'Ui label two',
                        type: ElementPropTypes.string
                    }
                })
            )
        },
        arrayTwo: {
            label: 'Array label',
            type: ElementPropTypes.arrayOf(
                ElementPropTypes.shape({
                    keyOne: {
                        label: 'Ui label one',
                        type: ElementPropTypes.string
                    },
                    keyTwo: {
                        label: 'Ui label two',
                        type: ElementPropTypes.string
                    }
                })
            )
        }
    };
```

Currently, using this system will cause the matching labels in the shapes in Site Designer to put the last shape keys under the first shape, and the other shapes will be empty.

Reply:
I'll write that up as a unit test before merging this so we know if that's fixed.

Reply:
@chris-decker-volusion confirmed your use case and covered with a unit test. Two things:

1. the old code would have failed because it would have used "Array label" as they key for arrayOne and arrayTwo.
2. it's nice to know that you can safely use the same keys deeper in a nested structure (in this example keyOne and keyTwo)